lib/omnibus/packagers/ips.rb

```diff
@@ -0,0 +1,193 @@
+      shellout!("/usr/bin/pkgmogrify -DARCH=`uname -p` #{staging_path("#{safe_base_package_name}.p5m.1")} #{staging_path('gen.manifestfile')} #{staging_path('doc-transform')} |pkgfmt > #{staging_path("#{safe_base_package_name}.p5m.2")}")
+    end
+
+    def generate_pkg_deps
+      shellout!("/usr/bin/pkgdepend generate -md #{source_dir} #{staging_path("#{safe_base_package_name}.p5m.2")}|pkgfmt > #{staging_path("#{safe_base_package_name}.p5m.3")}")
+      shellout!("/usr/bin/pkgmogrify -DARCH=`uname -p` #{staging_path("#{safe_base_package_name}.p5m.3")} #{staging_path('doc-transform')} |pkgfmt > #{staging_path("#{safe_base_package_name}.p5m.4")}")
+      shellout!("/usr/bin/pkgdepend resolve -m #{staging_path("#{safe_base_package_name}.p5m.4")}")
+    end
+
+    def check_pkg_manifest
+      shellout!("/usr/bin/pkglint -c /tmp/lint-cache -r http://pkg.oracle.com/solaris/release #{staging_path("#{safe_base_package_name}.p5m.4.res")}")
+    end
+
+    def create_ips_repo
+      shellout!("/usr/bin/pkgrepo create #{repo_dir}")
+    end
+
+    def publish_ips_pkg
+      shellout!("/usr/bin/pkgrepo add-publisher -s #{repo_dir} #{ENV['LOGNAME']}")
+      shellout!("/usr/bin/pkgsend publish -s #{repo_dir} -d #{source_dir} #{staging_path("#{safe_base_package_name}.p5m.4.res")}")
+    end
+
+    def view_repo_info
+      shellout!("/usr/bin/pkgrepo info -s #{repo_dir}")
+    end
+
+    def publish_as_pkg_archive
+      dest_dir = File.join(Config.package_dir, "#{safe_base_package_name}.p5p")
+      shellout!("/usr/bin/pkgrecv -s #{repo_dir} -a -d #{dest_dir} #{safe_base_package_name}")
+    end
+
+    # Return the IPS-ready base package name, converting any invalid characters to
+    # dashes (+-+).
+    #
+    # @return [String]
+    #
+    def safe_base_package_name
+      if project.package_name =~ /\A[a-z0-9\.\+\-]+\z/
+        project.package_name.dup
+      else
+        converted = project.package_name.downcase.gsub(/[^a-z0-9\.\+\-]+/, '-')
+
+        log.warn(log_key) do
+          "The `name' component of IPS package names can only include " \
+          "lowercase alphabetical characters (a-z), numbers (0-9), dots (.), " \
+          "plus signs (+), and dashes (-). Converting `#{project.package_name}' to " \
+          "`#{converted}'."
+        end
+        converted
+      end
+    end
+
+    # @see Base#package_name
+    #
+    def package_name
+      "#{safe_base_package_name}.p5p"
+    end
+
+    # For more info about fmri see:
+    # http://doc.oracle.com/cd/E23824_01/html/E21796/pkg-5.html#scrolltoc
+    def fmri_package_name
+      # TODO: still need to implement timestamp.
+      version = project.build_version.split(/[^\d]/)[1..2].join('.')
+      #version = project.build_version
+      "#{safe_base_package_name}@#{version},#{version}-#{project.build_iteration}"
+    end
+
+    #docfile = staging_path('doc-transform')
```

Comment:
Corrected the URL to http://docs.oracle.com/cd/E23824_01/html/E21796/pkg-5.html 
FMRI version for chef IPS package would look like:
=================
root@default-Solaris-113:~/pkg# pkgrepo list -s chef.p5p
PUBLISHER NAME                                          O VERSION
omnibus   developer/versioning/chef                       8.0,8.0-1:20160307T191823Z
=============